### PR TITLE
Fixes #156 use class name for validation

### LINKED ISSUE #156: Update validation to use class name instead of string

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: R/error-checks.R [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
+
+  stop(messages$errorWrongType(objectName, class(object)[1], objectTypes))
 }
 
+typeNamesFrom <- function(type){
+  if (is.character(type)) {
+    return(type)
+  }
+  type <- c(type)
+  sapply(type, function(t) t$classname)
```
Comment (msevestre) [inline R/error-checks.R]: This is just an extension that allows to either pass the class name or the class itself
e,g, validateIfOfType(obj, "Simulation") or validateIsOfType(obj, Simulation)

File: R/quantity.R [hunk 2]
```diff
@@ -42,7 +42,7 @@ Quantity <- R6::R6Class(
       private$printLine("Quantity Type", getEnumKey(QuantityType, self$quantityType))
     },
     printValue = function() {
-      private$printQuantityValue(self$name)
+      self$printQuantityValue(self$name)
```
Comment (msevestre) [inline R/quantity.R]: this was a bug

File: R/utilities-simulation.R [hunk 3]
```diff
@@ -112,7 +112,7 @@ resetSimulationCache <- function() {
 #' removeSimulationFromCache(sim1) # returns TRUE
 #' removeSimulationFromCache(sim2) # returns FALSE
 removeSimulationFromCache <- function(simulation) {
-  validateIsOfType(simulation, "Simulation")
+  validateIsOfType(simulation, Simulation)
```
Comment (msevestre) [inline R/utilities-simulation.R]: example of use. We can use either way. Just a convenience to be able to not rely on strings

Comment (PavelBal) [reply R/utilities-simulation.R]: Good idea. Let's merge it.

**Pull request merged**